acra/configuring-maintaining/installing/installing-from-github.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Installing from GitHub
+bookCollapseSection: true
+---
+
+### Installing from GitHub
+
+These are the instruction for installation of [AcraServer]({{< ref "acra/configuring-maintaining/general-configuration/acra-server.md#-INVALID" >}}) from the [Cossack Labs' GitHub repository for Acra](https://github.com/cossacklabs/acra/). You will need the same set of dependencies for each component.
+
+#### Installing the dependencies
+
+Install the dependencies for Acra:
+
+```
+sudo apt-get install git golang libssl-dev make build-essential
+```
+
+Set up your `$GOPATH` to some place where you will store the code.
+
+#### Install [Themis](https://www.github.com/cossacklabs/themis)
+
+```
+git clone https://github.com/cossacklabs/themis.git
+cd themis
+make
+sudo make install
+```
+
+#### Build the key generator and generate the keys
+
+```
+go get github.com/cossacklabs/acra/cmd/acra-keymaker
+```
+
+Then generate the [keys]({{< ref "acra/acra-in-depth/cryptography-and-key-management/#generating-all-the-acra-keys-in-one-go-INVALID" >}}) and distribute them across the infrastructure.
+
+Remember to generate `ACRA_MASTER_KEY` and assign it to the environmental variable!
+
+#### Set up the environment for AcraServer
+
+On a separate machine, create a user for AcraServer:
+```
+sudo useradd -m acra-server
+sudo su acra-server
+cd ~/
+export GOPATH=`pwd`
```

Comment:
explicit better than implicit. If someone familiar with golang then he knows how it works and can skip it. If some newbie reads and want to try then better we describe steps that will work in 100% cases in most of environments. Plus we should show path to binary below. If we don't set this variable then it will be `''` and command `$GOPATH/bin/acra-connector` will expanded to `""/bin/acra-connector`. And we need to explain that if you didn't set GOPATH then you should call binary from `$HOME/go/bin/executable`. If you have set then use `%OPATH/bin/executable` and bla bla.

Reply:
I'd still rather say once:

> Please make sure your GOBIN is in PATH:
> 
> ```
> export PATH=$PATH:${GOBIN:-${GOPATH:-$HOME}/go/bin}
> ```

and then proceed to use the same binary names that you would get if you installed Acra via packages.

Newbies who have no clue will just recite the spell in their terminal. People who have some Go experience will know where `go get` puts stuff and how to launch it.